lib/capybara/node/finders.rb

```diff
@@ -166,7 +166,13 @@ def all(*args)
       # @return [Capybara::Node::Element]            The found element or nil
       #
       def first(*args)
+        if Capybara.first_default_waiting
+          options = if args.last.is_a?(Hash) then args.pop.dup else {} end
+          args.push({minimum: 1}.merge(options))
+        end
         all(*args).first
+      rescue Capybara::ExpectationNotMet
+        nil
```

Comment:
To be clear: I can live with the current spec, was just wondering if this PR wouldn't break existing flow.

That being said, I'm not quite sure why `first` should return `nil`. You asked for a use case so here's the line of code that actually raised the original error:

```
 page.first('a', text: link).click
```

The error `undefined method 'click' for NilClass` is in my opinion less useable than `can't find element for selector 'a'`.